### ISSUE TITLE
Feature Suggestion

### ISSUE THREAD

**Issue by JabirSS** [martinjrobins/Aboria#35]
Dear @martinjrobins ,

I'd like to suggest adding the following features:

1- Being able to choose which variables to write out to VTK.
This is useful in case of simulations with a very large number of particles and number of variables that are not needed for visualization/post-processing, in order to reduce the size of files.

2- Being able to write time or other non-particle-associated variables to the VTK files.
This is quite useful for visualization and post-processing.
I currently write time to the VTK files by overloading the `vtkWriteGrid()` function in the Utils.h file as follows:

```C++
#ifdef HAVE_VTK
template <typename T=void>
void vtkWriteGrid(const char *name, int timestep, double time, vtkSmartPointer<vtkUnstructuredGrid> grid) {
    vtkSmartPointer<vtkXMLUnstructuredGridWriter> writer =
        vtkSmartPointer<vtkXMLUnstructuredGridWriter>::New();

#if VTK_MAJOR_VERSION > 5
    writer->SetInputData(grid);
#else
    writer->SetInput(grid);
#endif

    vtkDoubleArray *t = vtkDoubleArray::New();
    t->SetName("TIME");
    t->SetNumberOfTuples(1);
    t->SetTuple1(0, time);

    grid->GetFieldData()->AddArray(t);


    writer->SetDataModeToBinary();

    char buffer[100];
    sprintf(buffer,"%s%05d.vtu",name,timestep);

    writer->SetFileName(buffer);
    writer->Write();
}
#endif
```

This not might be the smartest or most efficient way to do it but it does what I need for now...

Thanks!

**Comment by martinjrobins** [comment]
yup, agree that both of these would be nice to have. 

For 2, how do you use the time values for visualisations? I know paraview has some time support, but I'm not too sure how it works. The way you overload vtkWriteGrid seems to indicate the you put the time values within each file you output. Do you have a file for each timestep of your simulation, or a single file for the entire simulation?

**Comment by JabirSS** [comment]
You're right. In my simulations I output VTK files periodically, that's why I pass the current time value to vtkWriteGrid.  One value (VTK calls it field data ) is written in each file so it practically takes no space at all.

In ParaView, I use AnnotateGlobalData to see the timestamp of the current file in my visualization window. 
It's also quite useful for time-series plots.

**Comment by martinjrobins** [comment]
ok, that makes sense. I'd be quite happy to just make those changes to the `vtkWriteGrid` function within Aboria. It doesn't make sense to add them to the main `Particles` data structure, as it is simply a collection of particles with no notion of time. But the `vtkWriteGrid` already has a notion of time (i.e. a timestep number), so it makes sense to be able to pass the actual value of time as well.

Feel free to create a Pull Request with this change and I can merge it in, or can make the change myself, up to you.

For 1, I'll have a look at how I can exclude given variables from the `copy_to_vtk_grid` function. option 1:  the user can supply a tuple of variables to exclude, option 2: I could create a separate function that "turns off" the output to vtk for a given variable"

**Comment by martinjrobins** [comment]
I ended up making number 2 a bit more general. The function now has the signature:

```cpp
/// Write a vtk unstructured grid to a filename of the form:
/// printf("name%05d",timestep). User can optionally provide \p fields, which is
/// a set of key/value pairs (e.g. "time"->0.1) written as additional fields to
/// the file
///
/// \param name the base filename
/// \param timestep the index in the filename
/// \param grid the grid to write to the file
/// \param fields a `std::map` of additional fields to write
///
template <typename T = void>
void vtkWriteGrid(const char *name, int timestep,
                  vtkSmartPointer<vtkUnstructuredGrid> grid,
                  const std::map<std::string, double> &fields = {});
 ```

so if you just want to add a time stamp to each file you would write

```cpp
vtkWriteGrid(name,timestep,grid,{{"time",the_time}});
```
And you can put in any number of additional fields with different names if you wish.

**Comment by JabirSS** [comment]
Excellent! Thanks a lot, this is quite useful!
